lib/core.js

```diff
@@ -803,7 +803,22 @@ Archiver.prototype.finalize = function() {
     this._finalize();
   }
 
-  return this;
+  var self = this:
+
+  return new Promise(function(resolve, reject) {
```

Comment:
instead of `self` we could use an arrow function (note sure node 4 supports it without the `--harmony` flag)